installer/util/brave_vpn_helper_utils.cc

```diff
@@ -0,0 +1,270 @@
+  std::wstring brave_vpn_entry_with_null(brave_vpn_entry);
+  brave_vpn_entry_with_null += L'\0';
+  // Allocate and set the SERVICE_TRIGGER_SPECIFIC_DATA_ITEM structure
+  SERVICE_TRIGGER_SPECIFIC_DATA_ITEM deviceData = {0};
+  deviceData.dwDataType = SERVICE_TRIGGER_DATA_TYPE_STRING;
+  // Exclude EOL
+  deviceData.cbData = brave_vpn_entry_with_null.size() *
+                      sizeof(brave_vpn_entry_with_null.front());
+  deviceData.pData = (PBYTE)brave_vpn_entry_with_null.c_str();
+  // Allocate and set the SERVICE_TRIGGER structure
+  SERVICE_TRIGGER serviceTrigger = {0};
+  serviceTrigger.dwTriggerType = SERVICE_TRIGGER_TYPE_CUSTOM;
+  serviceTrigger.dwAction = SERVICE_TRIGGER_ACTION_SERVICE_START;
+  serviceTrigger.pTriggerSubtype = const_cast<GUID*>(&kNetworkProfileGUID);
+  serviceTrigger.cDataItems = 1;
+  serviceTrigger.pDataItems = &deviceData;
+
+  // Allocate and set the SERVICE_TRIGGER_INFO structure
+  SERVICE_TRIGGER_INFO serviceTriggerInfo = {0};
+  serviceTriggerInfo.cTriggers = 1;
+  serviceTriggerInfo.pTriggers = &serviceTrigger;
+
+  // Call ChangeServiceConfig2 with the SERVICE_CONFIG_TRIGGER_INFO level
+  // and pass to it the address of the SERVICE_TRIGGER_INFO structure
+  return ChangeServiceConfig2(hService, SERVICE_CONFIG_TRIGGER_INFO,
+                              &serviceTriggerInfo);
+}
+
+}  // namespace
+
+bool ConfigureServiceAutoRestart(const std::wstring& service_name,
+                                 const std::wstring& brave_vpn_entry) {
+  ScopedScHandle scm(::OpenSCManager(nullptr, nullptr, SC_MANAGER_CONNECT));
+  if (!scm.IsValid()) {
+    LOG(ERROR) << "::OpenSCManager failed. service_name: " << service_name
+               << ", error: " << std::hex << HRESULTFromLastError();
+    return false;
+  }
+  ScopedScHandle service(
+      ::OpenService(scm.Get(), service_name.c_str(), SERVICE_ALL_ACCESS));
+  if (!service.IsValid()) {
+    LOG(ERROR) << "::OpenService failed. service_name: " << service_name
+               << ", error: " << std::hex << HRESULTFromLastError();
+    return false;
+  }
+
+  if (!brave_vpn::SetServiceFailureActions(service.Get())) {
+    LOG(ERROR) << "SetServiceFailureActions failed:" << std::hex
+               << HRESULTFromLastError();
+    return false;
+  }
+  if (!SetServiceTriggerForVPNConnection(service.Get(), brave_vpn_entry)) {
+    LOG(ERROR) << "SetServiceTriggerForVPNConnection failed:" << std::hex
+               << HRESULTFromLastError();
+    return false;
+  }
+  return true;
+}
+
+base::FilePath GetBraveVpnHelperServicePath() {
+  base::FilePath asset_dir;
+  base::PathService::Get(base::DIR_ASSETS, &asset_dir);
+  return asset_dir.Append(brave_vpn::kBraveVPNHelperExecutable);
+}
+
+// The service starts under system user so we save crashes to
+// %PROGRAMDATA%\BraveSoftware\{service name}\Crashpad
+base::FilePath GetVpnHelperServiceProfileDir() {
+  auto program_data = install_static::GetEnvironmentString("PROGRAMDATA");
```

Comment:
You could use:

```c++
std::wstring program_data = install_static::GetEnvironmentString(L"PROGRAMDATA");
```

And then you could avoid the `UTF8ToWide` conversion down below:
```c++
base::FilePath(program_data)
```